airflow/contrib/hooks/gcs_hook.py

```diff
@@ -185,40 +186,64 @@ def download(self, bucket_name, object_name, filename=None):
         else:
             return blob.download_as_string()
 
-    def upload(self, bucket_name, object_name, filename,
-               mime_type='application/octet-stream', gzip=False):
+    def upload(self, bucket_name, object_name, filename=None,
```

Comment:
Can you please add type annotations here? see the comment below - that would be much better than assertNone and also it's great for users of the hook.